File: go/problems/day18_test.go
package problems

import "testing"

var day18 Solvable[int] = Day18{}

func TestDay18_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day18_1")
	result := day18.Solve(input)

	var expected int = 62
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay18_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day18_2")
	result := day18.Solve(input)

	var expected int = 45159
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day18.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day18.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 18", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day18_1");
    const result = solve(input);
    assert.equal(result, 62);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day18_2");
    const result = solve(input);
    assert.equal(result, 45159);
  });
});
